controllers/sriovoperatorconfig_controller.go

```diff
@@ -309,3 +323,44 @@ func (r *SriovOperatorConfigReconciler) syncK8sResource(cr *sriovnetworkv1.Sriov
 	}
 	return nil
 }
+
+// syncSystemdService creates the Machine Config to deploy the systemd service on openshift ONLY
+func (r *SriovOperatorConfigReconciler) syncSystemdService(cr *sriovnetworkv1.SriovOperatorConfig) error {
+	logger := log.Log.WithName("syncSystemdService")
+	logger.Info("Start to sync config systemd machine config for openshift")
+
+	//TODO: list here the pools
+
+	data := render.MakeRenderData()
+	data.Data["LogLevel"] = cr.Spec.LogLevel
+	objs, err := render.RenderDir(constants.SystemdServiceOcpPath, &data)
+	if err != nil {
+		logger.Error(err, "Fail to render config daemon manifests")
+		return err
+	}
+
+	// Sync machine config
+	for _, obj := range objs {
+		if obj.GetKind() == "MachineConfig" && len(cr.Spec.ConfigDaemonNodeSelector) > 0 {
+			scheme := kscheme.Scheme
+			mc := &machinev1.ControllerConfig{}
+			err = scheme.Convert(obj, mc, nil)
+			if err != nil {
+				logger.Error(err, "Fail to convert to MachineConfig")
+				return err
+			}
+			mc.Labels = cr.Spec.ConfigDaemonNodeSelector
+			err = scheme.Convert(mc, obj, nil)
+			if err != nil {
+				logger.Error(err, "Fail to convert to Unstructured")
+				return err
+			}
```

Comment:
Seems like this could be a function with a name that tells us what it actually does without having to understand the code. something like `setLabelInsideObject`